src/Toml/Parser.hs

```diff
@@ -159,10 +161,19 @@ tableNameP = lexeme $ between (char '[') (char ']') keyP
 
 -- Values
 
+decimalP :: Parser Integer
+decimalP = mkInteger <$> decimalStringP
+  where
+    decimalStringP   = fst <$> match (some digitChar >> optional (try $ many _digitsP))
```

Comment:
The usage of `match` looks interesting 🤔It looks like and efficient enough way to get the part of the input. Well done!